src/Service/Telemetry/MyTelemetryInitializer.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using Azure.DataApiBuilder.Product;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+
+public class MyTelemetryInitializer : ITelemetryInitializer
+{
+    public static readonly IReadOnlyDictionary<string, string> GlobalProperties = new Dictionary<string, string>
+    {
+        { "UserAgent", $"{ProductInfo.GetDataApiBuilderUserAgent()}" },
+        // Add more custom properties here
+    };
+
+    public void Initialize(ITelemetry telemetry)
+    {
+        telemetry.Context.Cloud.RoleName = "DataApiBuilder";
+        telemetry.Context.Session.Id = Guid.NewGuid().ToString();
+        telemetry.Context.Component.Version = ProductInfo.GetProductVersion();
+        telemetry.Context.Device.OperatingSystem = Environment.OSVersion.ToString();
+        telemetry.Context.User.Id = $"{Environment.MachineName}_{Environment.UserName}";
```

Comment:
This assumes the user has these environment variables, right? This should be documented or exposed as options that the user can configure.